docs/proposals/20250724-independent-worker-topology-versions.md

```diff
@@ -0,0 +1,515 @@
+	//
+	// If not specified, the MachineDeployment will use the version from Cluster.Spec.Topology.Version.
+	// +optional
+	// +featureGate=IndependentWorkerTopologyVersion
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Version *string `json:"version,omitempty"`
+}
+```
+- We will also initially put this functionality behind a feature flag. 
+  - We will follow the same approach that the ClusterTopology flag takes. So, for example, if we create a cluster with the flag enabled, and then remove the flag from the controller, the controller webhooks would reject any version changes to the created cluster both for `topology.version` and `machinedeployment/machinepool.version`. 
+
+- What are some important details that didn't come across above.
+- What are the caveats to the implementation?
+- Go in to as much detail as necessary here.
+- Talk about core concepts and how they relate.
+
+### Security Model
+
+Document the intended security model for the proposal, including implications
+on the Kubernetes RBAC model. Questions you may want to answer include:
+
+* Does this proposal implement security controls or require the need to do so?
+  * If so, consider describing the different roles and permissions with tables.
+* Are their adequate security warnings where appropriate (see https://shostack.org/files/papers/ReederEtAl_NEATatMicrosoft.pdf for guidance).
+* Are regex expressions going to be used, and are their appropriate defenses against DOS.
+* Is any sensitive data being stored in a secret, and only exists for as long as necessary?
+
+### Risks and Mitigations
+
+- What are the risks of this proposal and how do we mitigate? Think broadly.
+- How will UX be reviewed and by whom?
+- How will security be reviewed and by whom?
+- Consider including folks that also work outside the SIG or subproject.
+
+## Alternatives
+
+The `Alternatives` section is used to highlight and record other possible approaches to delivering the value proposed by a proposal.
+
+## Upgrade Strategy
+
+If applicable, how will the component be upgraded? Make sure this is in the test plan.
+
+Consider the following in developing an upgrade strategy for this enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to make use of the enhancement?
+
+## Additional Details
+
+### Test Plan
+
+#### Scenario: Existing behavior with ClusterClass Topology Version
+When a cluster admin updates the `spec.topology.version` of their existing cluster with *no* worker node group versions defined, the default behavior should be verified. 
+- Ensure rolling upgrades of control planes and Workers are updated to the desired version. 
+
+#### Scenario: Single cluster uses Worker Node Group Version for the First Time
```

Comment:
so with the flip/flopping requirement it would be a new machinedeployment if we want to set it for the first time right?